webapp/components/notifications/NotificationMenu/NotificationMenu.vue

```diff
@@ -46,24 +48,53 @@ export default {
           variables,
         })
         if (!(markAsRead && markAsRead.read === true)) return
-        this.notifications = this.notifications.map(n => {
-          return n.from.id === markAsRead.from.id ? markAsRead : n
+        this.displayedNotifications = this.displayedNotifications.map(n => {
+          return this.equalNotification(n, markAsRead) ? markAsRead : n
         })
       } catch (err) {
-        throw new Error(err)
+        this.$toast.error(err.message)
       }
     },
+    equalNotification(a, b) {
+      return a.from.id === b.from.id && a.createdAt === b.createdAt && a.reason === b.reason
+    },
   },
   computed: {
-    totalNotifications() {
-      return (this.notifications || []).length
+    notificationsCount() {
+      return (this.displayedNotifications || []).length
+    },
+    unreadNotificationsCount() {
+      let countUnread = 0
+      if (this.displayedNotifications) {
+        this.displayedNotifications.forEach(notification => {
+          if (!notification.read) countUnread++
+        })
+      }
+      return countUnread
     },
   },
   apollo: {
     notifications: {
       query() {
         return notificationQuery(this.$i18n)
       },
+      pollInterval() {
+        return NOTIFICATIONS_POLL_INTERVAL
+      },
+      update(data) {
+        const newNotifications = data.notifications.filter(newN => {
+          return !this.displayedNotifications.find(oldN => this.equalNotification(newN, oldN))
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-23T10:54:58Z" title="Monday, September 23rd 2019, 12:54:58 pm +02:00">Sep 23, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -58,6 +104,15 @@ export default {
     totalNotifications() {
       return (this.notifications || []).length
     },
+    unredNotifications() {
```


```suggestion
    unreadNotifications() {
```